### PR TITLE
Some edits of the KINSol documentation.

### PR DESCRIPTION
**Pull request by singima**

### CODE REVIEW

File: include/deal.II/sundials/kinsol.h [hunk 1]
```diff
@@ -161,7 +161,9 @@ namespace SUNDIALS
    *
    * Specifying residual() allows the user to use Newton and Picard strategies
    * (i.e., $F(u)=0$ will be solved), while specifying iteration_function(), a
-   * fixed point iteration will be used (i.e., $G(u)=u$ will be solved).
+   * fixed point iteration will be used (i.e., $G(u)=u$ will be solved). An
+   * error will be thrown if iteration_function() is called for Picard or
```
Comment (tjhei) [inline include/deal.II/sundials/kinsol.h:165]:
```suggestion
   * error will be thrown if iteration_function() is set for Picard or
```
is that what you mean?

File: include/deal.II/sundials/kinsol.h [hunk 2]
```diff
@@ -292,11 +294,11 @@ namespace SUNDIALS
       add_parameters(ParameterHandler &prm);
 
       /**
-       * The solution strategy to use. If you choose SolutionStrategy::newton
-       * or SolutionStrategy::linesearch, you have to provide also the function
-       * residual(). If you choose SolutionStrategy::picard or
+       * The solution strategy to use. If you choose SolutionStrategy::newton,
+       * SolutionStrategy::linesearch, or SolutionStrategy::picard you have to
+       * provide also the function residual(). If you choose
```
Comment (tjhei) [inline include/deal.II/sundials/kinsol.h:299]:
```suggestion
       * provide the function residual(). If you choose
```

File: include/deal.II/sundials/kinsol.h [hunk 3]
```diff
@@ -418,8 +420,8 @@ namespace SUNDIALS
     /**
      * A function object that users should supply and that is intended to
      * compute the residual `dst = F(src)`. This function is only used if the
-     * SolutionStrategy::newton or SolutionStrategy::linesearch strategies
-     * were selected.
+     * SolutionStrategy::newton, SolutionStrategy::linesearch, or
+     * SolutionStrategy::newton strategies were selected.
```
Comment (tjhei) [inline include/deal.II/sundials/kinsol.h:424]: do you mean picard here?

File: include/deal.II/sundials/kinsol.h [hunk 4]
```diff
@@ -434,8 +436,7 @@ namespace SUNDIALS
      * A function object that users should supply and that is intended to
      * compute the iteration function $G(u)$ for the fixed point and Picard
      * iteration. This function is only used if the
-     * SolutionStrategy::fixed_point or SolutionStrategy::picard strategies
-     * were selected.
+     * SolutionStrategy::picard strategy is selected.
```
Comment (tjhei) [inline include/deal.II/sundials/kinsol.h:439]: do you mean fixed_point?

**Review by tjhei — COMMENTED** [review]
Thanks, Sean!

File: include/deal.II/sundials/kinsol.h (hunk 4 above)
Comment (singima) [reply include/deal.II/sundials/kinsol.h]: Made the edits and squashed

File: include/deal.II/sundials/kinsol.h [hunk 5]
```diff
@@ -434,8 +436,7 @@ namespace SUNDIALS
      * A function object that users should supply and that is intended to
      * compute the iteration function $G(u)$ for the fixed point and Picard
```
Comment (tjhei) [inline include/deal.II/sundials/kinsol.h:437]: What about the "Picard" here?

File: include/deal.II/sundials/kinsol.h [hunk 6]
```diff
@@ -292,11 +294,11 @@ namespace SUNDIALS
...
-       * The solution strategy to use. If you choose SolutionStrategy::newton
-       * or SolutionStrategy::linesearch, you have to provide also the function
-       * residual(). If you choose SolutionStrategy::picard or
-       * SolutionStrategy::fixed_point, you have to provide also the function
-       * iteration_function().
+       * The solution strategy to use. If you choose SolutionStrategy::newton,
+       * SolutionStrategy::linesearch, or SolutionStrategy::picard you have to
+       * provide also the function residual(). If you choose
+       * SolutionStrategy::fixed_point, you have to provide the function
+       *  iteration_function().
```
Comment (tjhei) [inline include/deal.II/sundials/kinsol.h:301]:
```suggestion
       * iteration_function().
```

File: include/deal.II/sundials/kinsol.h (hunk 2 above)
Comment (tjhei) [reply include/deal.II/sundials/kinsol.h:299]: what about this suggestion?

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
Great, thank you!

**Pull request merged**